### PR TITLE
MailChimp API key input.

### PR DESCRIPTION
**Pull request by budzanowski**
### Information

This is a part of #17434 PR that represents MailChimp API key input UI view. This is the second step of the setup procedure.
This is provided for review as a separate PR because the original (spike) PR is too big for the review to be comfortable. 

To test this component in action please refer to #17434.

image

image

### CODE REVIEW

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+			<div>{ translate( 'From there, grab an existing key or generate a new on for your store.' ) } </div>
+		</div>
+		<FormLabel required={ ! isKeyCorrect }>
+			{ translate( 'Mailchimp API Key:' ) }
+		</FormLabel>
+		<FormTextInput
+			name={ 'api_key' }
+			isError={ ! isKeyCorrect }
+			placeholder={ 'Enter your MailChimp API key' }
+			onChange={ onChange }
```
Comment (timmyc) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js]: Perhaps add a default propType, or a check to make sure `onChange` is defined here.

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	<FormFieldset className="setup-steps__mailchimp-key-input">
+		<div className="setup-steps__mailchimp-api-intro-notice">
+			{ translate( 'Now that you\'re signed in to MailChimp, you need an API key to start the connection process' ) }
+		</div>
+		<div className="setup-steps__mailchimp-api-directions" >
+			<span>{ translate( 'To find your Mailchimp API key, go to ' ) }</span>
+			<span className="setup-steps__mailchimp-api-directions-bold">
+				{ translate( 'Settting > Extras > API keys.' ) }
+			</span>
+			<div>{ translate( 'From there, grab an existing key or generate a new on for your store.' ) } </div>
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js]: Typo. `on` -> `one`.

`'From there, grab an existing key or generate a new one for your store.`

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+export default localize( ( { translate, onChange, apiKey, isKeyCorrect } ) => (
+	<FormFieldset className="setup-steps__mailchimp-key-input">
+		<div className="setup-steps__mailchimp-api-intro-notice">
+			{ translate( 'Now that you\'re signed in to MailChimp, you need an API key to start the connection process' ) }
+		</div>
+		<div className="setup-steps__mailchimp-api-directions" >
+			<span>{ translate( 'To find your Mailchimp API key, go to ' ) }</span>
+			<span className="setup-steps__mailchimp-api-directions-bold">
+				{ translate( 'Settting > Extras > API keys.' ) }
```
Comment (justinshreve) [inline client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js]: the `To find your Mailchimp API key, go to Settting > Extras > API keys.` copy needs changed a bit. There is no Settings section.

Maybe something like

 `To find your Mailchimp API key, click your profile picture, select 'Account', and go to Extras > API keys.`

File: client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js (hunk 1 above)
Comment (budzanowski) [reply client/extensions/woocommerce/app/settings/email/mailchimp/setup-steps/key-input.js]: isRequired.

**Comment by budzanowski** [comment]
PropTypes added and wording changed. Good to go?

**Pull request merged**